mainargs/src-3/Macros.scala

```diff
@@ -0,0 +1,155 @@
+package mainargs
+
+import scala.quoted._
+
+object Macros {
```

Comment:
It's often a good pattern to have macros in a class that itself takes a context parameter `Quotes`. Then it is easier (and clearer) to use helper functions inside the class since you don't have to pass `Quotes` to every method.
Not saying that it's needed here, just a suggestion :smiley:

Reply:
Do you have an example? Because I tried that at the compiler complains since the macro should be places in a stable place (like an object).

Reply:
You can take a look here. It's a draft, but this part works.

https://github.com/softwaremill/magnolia/blob/03abe6b81c546c35ece084cb45b8967212d1dff9/src/core/magnolia.scala#L61

Reply:
Since I need to have a object either way, and all the other repositories in `com-lihaoyi` use this pattern, I'll leave it as it is.